R/formula.R

```diff
@@ -0,0 +1,422 @@
+library(stringr)
+
+extract_terms_response <- function(formula) {
+  tt <- terms(formula)
+  terms <- attr(tt, "term.labels")
+  variables <- as.list(attr(tt, "variables")[-1])
+  response <- attr(tt, "response")
+  global_intercept <- attr(tt, "intercept") == 1
+
+  hier <- str_match(terms, "\\|")
+  int <- str_match(terms, ":")
+  group_terms <- terms[!is.na(hier)]
+  int_terms <- terms[!is.na(int) & is.na(hier)]
+  individual_terms <- terms[is.na(hier) & is.na(int)]
+
+  if (response)
+    response <- variables[response]
+  else
+    response <- NA
+
+  list(individual_terms=individual_terms,
+       int_terms=int_terms,
+       group_terms=group_terms,
+       response=extract_response(response),
+       global_intercept=global_intercept)
+}
+
+extract_response <- function(response) {
```

Comment:
These functions can be a little bit tricky so I will explain it here as a reference for when we talk in person. 

When you call `terms(form)` the response is an unevaluated object, it's not a string. At some points in projpred the response is actually a list of responses, where we are basically fitting the same model to different draws from the posterior for the parameters. In this case we want to split the formula to individual formulas so that lmer can fit them. Therefore, we leverage the splitting of these terms to several functions. This is the one that takes an unevaluated response object and returns a vector of strings, or a single string if it represents just a single response.